python/ray/util/sgd/v2/worker_group.py

```diff
@@ -77,16 +85,31 @@ def __init__(self,
         self.num_gpus_per_worker = num_gpus_per_worker
         self.additional_resources_per_worker = additional_resources_per_worker
         self.workers = []
+        if actor_cls:
+            if not issubclass(actor_cls, BaseWorkerMixin):
+                raise ValueError(f"The provided remote class {actor_cls} "
+                                 f"must extend `BaseWorkerMixin`")
+            self._base_cls = actor_cls
+        else:
+            self._base_cls = BaseWorkerMixin
+
+        self._actor_cls_args = actor_cls_args or []
+        self._actor_cls_kwargs = actor_cls_kwargs or {}
+
         # TODO(matt): Validate resources. Fast-fail if it is impossible to
         #  handle the request, rather than hang indefinitely.
         self._remote_cls = ray.remote(
             num_cpus=self.num_cpus_per_worker,
             num_gpus=self.num_gpus_per_worker,
-            resources=self.additional_resources_per_worker)(BaseWorker)
+            resources=self.additional_resources_per_worker)(self._base_cls)
         self.start()
 
     def _create_worker(self):
-        return self._remote_cls.remote()
+        if type(self._base_cls) == BaseWorkerMixin:
```

Comment:
Just double checking my Python understanding, there's nothing wrong with initializing a `Mixin` right?

Reply:
Correct, this is just a standard Python class.